src/v/cluster/partition_balancer_backend.cc

```diff
@@ -358,9 +362,20 @@ ss::future<> partition_balancer_backend::do_tick() {
     // status requests by default 700ms
     const auto node_responsiveness_timeout = _node_status_interval() * 7;
 
+    auto active_mode = _mode();
+
+    const auto& loaded_license = _feature_table.get_license();
+    const bool has_valid_license = loaded_license.has_value()
+                                   && !loaded_license->is_expired();
+    if (
+      active_mode == model::partition_autobalancing_mode::continuous
+      && !has_valid_license) {
+        // TODO(ik) add logging/warning
```

Comment:
If we simply add a WARN log here, it will be printed on every tick period.

Would that be a feature or a bug?

Reply:
I vote feature. The default interval is 30s  https://github.com/redpanda-data/redpanda/blob/fb388a0fcc98ba029e2d33b36b13ad19cc6334ac/src/v/config/configuration.cc#L3003-L3008

which seems not too horrible for what amounts to a license nag.

Reply:
> If we simply add a WARN log here, it will be printed on every tick period.

What about on construction/initialization and on change?

> Would that be a feature or a bug?

We already have a log nag every 5mins, which I believe now logs which feature.

Reply:
> What about on construction/initialization and on change?

My understanding is that we cannot initialise/change a property to a restricted value when there isn't a license. @oleiman is my reading of the enterprise property correct?

Reply:
>cannot initialize/change a property

Sort of. The intent of the enterprise property wrapper is to provide the exact same API contract as a regular property. The responsibility of avoiding "restricted" values is offloaded to the API consumer, and `enterprise<P>::set_value` itself is behaviorally indistinguishable from `property<T>::set_value`.

Put another way, the only differences between an enterprise property and any other are the ability to specify a restricted condition in the constructor AND the ability to check whether some proposed setting violates that condition.

Reply:
Worth adding that a "restricted" condition doesn't represent a violation by itself. It is only a violation when coupled with an invalid license.